pkg/providers/pricing/controller.go

```diff
@@ -50,9 +52,13 @@ func (c *Controller) Builder(_ context.Context, m manager.Manager) corecontrolle
 
 func (c *Controller) updatePricing(ctx context.Context) error {
 	work := []func(ctx context.Context) error{
-		c.pricingProvider.UpdateOnDemandPricing,
 		c.pricingProvider.UpdateSpotPricing,
 	}
+
+	if !options.FromContext(ctx).IsolatedVPC {
```

Comment:
nit: Worth a comment over this to describe why we can fetch spot pricing but not on-demand pricing?